spec/features/password_reset_spec.rb

```diff
@@ -0,0 +1,81 @@
+
+feature 'password can be reset' do
+
+  before do
+    sign_up
+    Capybara.reset!
+    allow(SendRecoverLink).to receive(:call)
+  end
+
+  let(:user) { User.first }
+
+  scenario 'users can click "forgotten password" link' do
+    visit '/sessions/new'
+    click_link 'Forgotten password?'
+    expect(page).to have_content "Please enter your email address"
+  end
+
+  scenario 'users can enter email address' do
+    recover_password
+    expect(page).to have_content 'Thank you, please check your inbox'
+  end
+
+  scenario 'users receive a reset token on sign up' do
+    expect{recover_password}.to change {User.first.password_token}
+  end
+
+  scenario 'tokens expire after an hour' do
+    recover_password
+    Timecop.travel(60*60*60) do
+      visit("/users/reset_password?token=#{user.password_token}")
+      expect(page).to have_content "Your token has expired"
+    end
+  end
+
+  scenario 'user may reset passwords' do
+    recover_password
+    visit("/users/reset_password?token=#{user.password_token}")
+    expect(page).to have_content "Please enter your new password"
+    end
```

Comment:
Lint/BlockAlignment: `end` at 39, 4 is not aligned with `scenario 'user may reset passwords' do` at 35, 2.